samples/security/TokenLoginModule.java

```diff
@@ -0,0 +1,256 @@
+/**
+ * Copyright (c) 2007-2017, Kaazing Corporation. All rights reserved.
+ */
+
+package com.kaazing.demo.loginmodules;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.security.auth.Subject;
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import javax.security.auth.login.LoginException;
+import javax.security.auth.spi.LoginModule;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.kaazing.gateway.server.spi.security.AuthenticationToken;
+import org.kaazing.gateway.server.spi.security.AuthenticationTokenCallback;
+import org.kaazing.gateway.server.spi.security.ExpiringState;
+import org.kaazing.gateway.server.spi.security.LoginModuleOptions;
+import org.kaazing.gateway.server.spi.security.LoginResult;
+import org.kaazing.gateway.server.spi.security.LoginResultCallback;
+
+// This is an example custom login module that reads a token to extract the username. If
```

Comment:
As far as I know, Javadoc is also used for implementations. If is a Kaazing practice, then I can live with that.